code/datums/martial/monk.dm

```diff
@@ -0,0 +1,455 @@
+				make_disciple.Grant(owner)
+				make_disciple.uses++
+		if(25)
+			if(!is_disciple)
+				to_chat(owner, "<span class = 'danger'>You can now make your second Disciple.</span>")
+				make_disciple.uses++
+		if(30)
+			if(!is_disciple)
+				to_chat(owner, "<span class = 'danger'>You can now make your final Disciple.</span>")
+				make_disciple.uses++
+
+/datum/martial_art/monk/proc/attack_roll(mob/living/T, abm)
+	if(istype(T, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = T
+		if(H.mind && istype(H.mind.martial_art, /datum/martial_art/monk))
+			var/datum/martial_art/monk/M = H.mind.martial_art
+			return M.defense_roll(curr_ab + abm)
+	var/armor_class = 10 + (T.getarmor("chest", "melee") * 0.1)
+	var/attack_bonus = curr_ab + abm
+	if(T.incapacitated())
+		attack_bonus += 3
+	var/attack_roll = roll(1,20) + attack_bonus
+	if(attack_roll >= armor_class)
+		if(attack_roll >= 20)
+			attack_roll = roll(1,20) + attack_bonus
+			if(attack_roll >= armor_class)
+				return ATK_CRIT_HIT
+			return ATK_HIT
+		else
+			return ATK_HIT
+	return ATK_MISS
+
+/datum/martial_art/monk/proc/defense_roll(abm)
+	var/armor = (owner.getarmor("chest", "melee") * 0.1)
+	var/armor_class = 10
+	if(!armor)
+		armor_class = 15
+	else
+		armor_class -= armor
+	var/attack_bonus = abm
+	if(owner.incapacitated())
+		attack_bonus += 3
+	var/attack_roll = roll(1,20) + attack_bonus
+	if(attack_roll >= armor_class)
+		if(attack_roll >= 20)
+			attack_roll = roll(1,20) + attack_bonus
+			if(attack_roll >= armor_class)
+				return ATK_CRIT_HIT
+			return ATK_HIT
+		else
+			return ATK_HIT
+	return ATK_MISS
+
+/datum/martial_art/monk/proc/add_exp(amt)
+	if(current_level == level_cap)
+		return
+	current_exp += amt
+	if(current_exp >= next_level_exp)
+		current_level++
+		var/next_level = current_level + 1
+		next_level_exp = next_level*100
+		do_level_up()
+		to_chat(owner, "<span class = 'danger'>You feel more confident in your powers.</span>")
+
+/datum/martial_art/monk/proc/do_attack(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D, use_cleave, use_circle, surrounding_mobs)
+	var/picked_hit_type = pick("punched")
+	switch(picked_hit_type)
+		if("punched")
+			A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	var/atr = attack_roll(D, flurry_penalty())
+	if(atr)
+		var/old_stat = D.stat
+		if(D.stat != DEAD && D.ckey && D != A)
+			add_exp(5)
+		var/dmg = rand(HIT_MIN, HIT_MAX)
+		var/was_crit = ""
+		var/armor = (D.run_armor_check("chest", "melee", armour_penetration = ki_level)) / 100
+		if(armor > 1)
+			armor = 1
+		if(atr == ATK_CRIT_HIT)
+			dmg *= 2
+			was_crit = "critically "
+		dmg -= (dmg * armor)
+		D.adjustBruteLoss(dmg)
+		if(use_circle)
+			do_attack(A, pick(surrounding_mobs), FALSE, FALSE, null)
+		if(old_stat != DEAD && D.stat == DEAD)
+			if(use_cleave)
+				switch(use_cleave)
+					if(1)
+						do_attack(A, pick(surrounding_mobs), FALSE, FALSE, null)
+					if(2)

```

Comment:
swap these magic numbers for defines

Reply:
Did some research these few days and don't quite know how to do what you want unfortunately. I'll see if I can contact some people or ask from help to see if there's a solution.

Reply:
Thanks to some outside help I've found a idiotically simple answer to the problem